tests/integration/vpp/040_acl_test.go

```diff
@@ -0,0 +1,906 @@
+	} else {
+		t.Logf("Configured IP is present.")
+
+	}
+	if isForInterface {
+		t.Logf("dumped acl is correctly assigned to interface %v", ifName2)
+	} else {
+		t.Fatalf("dumped interface is not correctly assigned")
+	}
+
+	// negative test
+	err = h.ModifyACL(1, aclErr1Rules, "test_modify1")
+	Expect(err).To(Not(BeNil()))
+	t.Logf("modifying of acl failed: %v", err)
+
+	const aclname3 = "test_modify2"
+	err = h.ModifyACL(1, aclNoRules, aclname3)
+	Expect(err).To(BeNil())
+	t.Logf("acl was modified")
+
+	acls, errx = h.DumpACL()
+	Expect(errx).To(BeNil())
+	aclCnt = len(acls)
+	Expect(aclCnt).Should(Equal(1))
+	t.Logf("%d acls dumped", aclCnt)
+
+	isPresent = false
+	isForInterface = false
+	for _, item := range acls {
+		if item.Meta.Index == aclIdx { //&& (aclname2 == item.Meta.Tag) {
+			t.Logf("Found modified ACL aclIPrules with aclName  %v", item.Meta.Tag)
+		}
+		if item.ACL.String() == modifiedacl.ACL.String() {
+			t.Logf("Last update caused no change in acl definition.")
+			break
+		} else {
+			t.Fatalf("Last update caused change in acl definition.")
+		}
+	}
+
+	// DELETE ACL
+	err = h.RemoveACLFromInterfacesAsEgress(aclIdx, []uint32{ifIdx2})
+
+	acls, errx = h.DumpACL()
+	Expect(errx).To(BeNil())
+	aclCnt = len(acls)
+	Expect(aclCnt).Should(Equal(1))
+	t.Logf("%d acls dumped", aclCnt)
+
+	isPresent = false
+	isForInterface = false
+	for _, item := range acls {
+		if item.Meta.Index == aclIdx { //&& (aclname2 == item.Meta.Tag) {
+			t.Logf("Found modified ACL aclIPrules with aclName  %v", item.Meta.Tag)
+		}
+		if item.ACL.Interfaces.String() == "" {
+			t.Logf("Interface assignment was removed")
+		} else {
+			t.Fatalf("Interface assignment was not removed.")
+		}
+	}
+
+	err = h.DeleteACL(aclIdx)
+	Expect(err).To(BeNil())
+	t.Logf("deleting acls succeed")
+
+	acls, errx = h.DumpACL()
+	Expect(errx).To(BeNil())
+	aclCnt = len(acls)
+	Expect(aclCnt).Should(Equal(0))
+	t.Logf("%d acls dumped", aclCnt)
+
+}
+
+// Test add MACIP acl rules
+func TestCRUDMacIPAcl(t *testing.T) {
+	ctx := setupVPP(t)
+	defer ctx.teardownVPP()
+
+	ih := ifplugin_vppcalls.CompatibleInterfaceVppHandler(ctx.vppBinapi, logrus.NewLogger("test"))
+
+	const ifName = "loop1"
+	ifIdx, errI := ih.AddLoopbackInterface(ifName)
+	Expect(errI).To(BeNil())
+	t.Logf("interface created %v", ifIdx)
+
+	const ifName2 = "loop2"
+	ifIdx2, errI2 := ih.AddLoopbackInterface(ifName2)
+	Expect(errI2).To(BeNil())
+	t.Logf("interface created %v", ifIdx2)
+
+	ifIndexes := ifaceidx.NewIfaceIndex(logrus.NewLogger("test-iface1"), "test-iface1")
+	ifIndexes.Put(ifName, &ifaceidx.IfaceMetadata{
+		SwIfIndex: ifIdx,
+	})
+	ifIndexes.Put(ifName2, &ifaceidx.IfaceMetadata{
+		SwIfIndex: ifIdx2,
+	})
+
+	h := aclplugin_vppcalls.CompatibleACLVppHandler(ctx.vppBinapi, ifIndexes, logrus.NewLogger("test"))
+	if h == nil {
+		t.Fatalf("handler was not created")
+	}
+
+	acls, errx := h.DumpMACIPACL()
+	Expect(errx).To(BeNil())
+	aclCnt := len(acls)
+	Expect(aclCnt).Should(Equal(0))
```

Comment:
You can use [`HaveLen(INT)`](https://onsi.github.io/gomega/#havelencount-int) matcher instead of this.

```go
Expect(acls).Should(HaveLen(0))
// in this case even
Expect(acls).Should(BeEmpty())
```

Reply:
Thanks for advice